partner_pricelist_search/models/res_partner.py

```diff
@@ -0,0 +1,33 @@
+# Copyright 2021 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class Pricelist(models.Model):
+    _inherit = "res.partner"
+
+    property_product_pricelist = fields.Many2one(
+        search="_search_property_product_pricelist")
+
+    @api.model
+    def _search_property_product_pricelist(self, operator, value):
+        if operator == "=":
+            def filter_func(partner):
+                return partner.property_product_pricelist.id == value
```

Comment:
For clearer code, you can do here lambda function:
```suggestion
            filter_func = lambda x: x.property_product_pricelist.id == value
```

Reply:
Had used lambda:
https://github.com/OCA/partner-contact/compare/79ea65cd67e7f3282e6c44ac95c7beb07db168cd..cf47c5e3cdca8e5b6563a927f8a4670607b3c075
But travis turned red for PEP8 E731:
https://travis-ci.com/github/OCA/partner-contact/jobs/472143017#L322-L326

Reply:
OK, I see https://www.flake8rules.com/rules/E731.html as reasoning for tracebacks, but don't really agree with the resulting code for such simple thing.